node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/InfraDeployerImpl.java

```diff
@@ -51,9 +51,19 @@ public InfraDeployerImpl(NodeRepository nodeRepository, Provisioner provisioner,
     }
 
     @Override
-    public Map<ApplicationId, Deployment> getSupportedInfraDeployments() {
-        return duperModel.getSupportedInfraApplications().stream()
-                .collect(Collectors.toMap(InfraApplicationApi::getApplicationId, InfraDeployment::new));
+    public void activateAllSupportedInfraApplications() {
+        duperModel.getSupportedInfraApplications().forEach(api -> {
+            var application = api.getApplicationId();
+            var deployment = new InfraDeployment(api);
+            try {
+                deployment.activate();
+            } catch (RuntimeException e) {
+                logger.log(LogLevel.INFO, "Failed to activate " + application, e);
+                // loop around to activate the next application
+            }
+        });
+
+        duperModel.infraApplicationsIsNowComplete();
```

Comment:
What if `activate()` throws?

Reply:
I'm not sure. On one hand we do not want to mark the duper model as complete. But on the other hand, it would be wrong to start the config server without the required infra apps. But failing construction might also be a bit dangerous if the failure is triggered by zk data it is hard to inspect as the zk server is brought down if the config server goes down.

So we should trigger deploy in InfrastructureProvisioner constructor and let RuntimeException propagate there (but not in maintain()) after a delay of say 1m, to fail config server startup?